module/PowerShellEditorServices/Commands/Public/CmdletInterface.ps1

```diff
@@ -36,18 +36,17 @@ function Register-EditorCommand {
     {
         $commandArgs = @($Name, $DisplayName, $SuppressOutput.IsPresent)
 
-        if ($ScriptBlock -ne $null)
+        $editorCommand = if ($ScriptBlock -ne $null)
         {
-            Write-Verbose "Registering command '$Name' which executes a ScriptBlock"
-            $commandArgs += $ScriptBlock
+            $null = Write-Verbose "Registering command '$Name' which executes a ScriptBlock"
+            [Microsoft.PowerShell.EditorServices.Extensions.EditorCommand, Microsoft.PowerShell.EditorServices]::new($Name, $DisplayName, $SuppressOutput, $ScriptBlock)
         }
         else
         {
-            Write-Verbose "Registering command '$Name' which executes a function"
-            $commandArgs += $Function
+            $null = Write-Verbose "Registering command '$Name' which executes a function"
```

Comment:
why the `$null =`?

Reply:
Just as a guarantee that `Write-Verbose` won't emit any arguments. But if we already feel like that's guaranteed, I can take it out

Reply:
Yeah it doesn't call `WriteObject`